docs/modules/nonpareil.md

```diff
@@ -0,0 +1,25 @@
+---
+name: Nonpareil
+url: https://github.com/lmrodriguezr/nonpareil
+description: >
+  Estimate metagenomic coverage and sequence diversity.
+---
+
+Nonpareil uses the redundancy of the reads in a metagenomic dataset to estimate
+the average coverage and predict the ammount of sequences that will be required
+to achieve "nearly complete coverage", defined as ≥95% or ≥99% average coverage.
+
+Since Nonpareil main output has no model information, it is necessary to run its
+auxiliary `R` plot functions and save the `curves` object as a `JSON` file. For
+more info on how to generate this file, check `nonpareil` [snakemake plot wrapper](https://snakemake-wrappers.readthedocs.io/en/latest/wrappers/nonpareil/plot.html#code). Briefly, call function `export_curve()` on object `curves`:
```

Comment:
What I meant, is that it would be helpful to give brief instructions how to run the _snakemake wrapper_ (or `RScript` in command line if it's easier to just run those two lines)

Reply:
There is no need to use `snakemake`, as the users can still use `nonpareil` and make the plots by themselves. The only extra thing they have to do, is to save the `R` object into a `json` file. Since I already have some functions for that implemented in a `snakemake` wrapper, I thought it would just be easier (instead of each user having to figure out how to parse the `R` object).

Reply:
If I understand you correctly, the functions in question are pretty short - is that right?

eg. it's this bit?
```R
  base::library("jsonlite")
  base::message("Exporting model as JSON")

  export_curve <- function(object){
    # Extract variables
    n <- names(attributes(object))[c(1:12,21:29)]
    x <- sapply(n, function(v) attr(object,v))
    names(x) <- n
    # Extract vectors
    n <- names(attributes(object))[13:20]
    y <- lapply(n, function(v) attr(object,v))
    names(y) <- n
    # Add model
    # https://github.com/lmrodriguezr/nonpareil/blob/162f1697ab1a21128e1857dd87fa93011e30c1ba/utils/Nonpareil/R/Nonpareil.R#L330-L332
    x_min <- 1e3
    x_max <- signif(tail(attr(curves$np.curves[[1]],"x.adj"), n=1)*10, 1)
    x.model <- exp(seq(log(x_min), log(x_max), length.out=1e3))
    y.model <- predict(object, lr=x.model)

    c(x, y, list(x.model=x.model), list(y.model=y.model))
  }

  export_set <- function(object){
    y <- lapply(object$np.curves, "export_curve")
    names(y) <- sapply(object$np.curves, function(n) n$label)
    jsonlite::prettify(toJSON(y, auto_unbox=TRUE))
  }

  y <- export_set(curves)
  write(y, snakemake@output[["json"]])
  base::message("JSON file saved")
```

I try to keep dependencies to 3rd party websites and the like to a minimum where possible. My preference would be to either include the above code in the docs here, or as a script in the `nonpareil` repo itself (seeing as the MultiQC module already depends on that). Either option would feel a little more resistant to URLs changing over time.

Reply:
I am not the owner of the `nonpareil` repo and not sure how active it still is, but I can definitely include the code in the docs.